### DISCUSSION TITLE
IntelliSense configuration

### DISCUSSION THREAD

**Discussion by dreamawakening** [tailwindlabs/headlessui#815] [Ideas]
Headless UI should include an IntelliSense configuration for VSCode by default.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your question!
Much appreciated! 🙏

I'm confused what you mean exactly. Can you elaborate on what you want specifically?
Headless UI is a JS library written in TypeScript, meaning that vscode should already provide you intellisense and code completions.

If you are talking about usage with tailwindcss, then that should also already work, since code completion for tailwind works in `class` and `className` props already.

**Reply by dreamawakening** [reply]
@RobinMalfait sorry I specifically meant using TailwindCSS IntelliSense with `Transition` components:

```tsx
<Transition
  show={editing}
  enter="transition-opacity transition-transform duration-75"
  enterFrom="opacity-0 -translate-y-5"
  enterTo="opacity-100 translate-y-0"
  leave="transition-opacity transition-transform duration-150"
  leaveFrom="opacity-100 translate-y-0"
  leaveTo="opacity-0 -translate-y-5"
>
  children
</Transition>
```

These props aren't `className`, so it doesn't work.

I tried adding a custom config for this:

```jsonc
{
  "tailwindCSS.experimental.classRegex": [
    "enter=\"([^\"]*)", // <div enter="..." />
    // ...
  ]
}
```

But that didn't work either.

**Comment by MichaelAllenWarner** [comment]
Some "official" recommended way of supporting this would indeed be nice.

My solution is to do this in `.vscode/settings.json`:

```json
{
  "tailwindCSS.experimental.classRegex": [
    "/\\* tw \\*/ '([^']*)",
    "/\\* tw \\*/ `([^`]*)",
    "/\\* tw \\*/ \"([^\"]*)"
  ]
}
```

and then use syntax like this in the JSX:

```jsx
<Transition.Child
  enter={/* tw */ 'ease-out duration-300'}
  enterFrom={/* tw */ 'opacity-0'}
  enterTo={/* tw */ 'opacity-100'}
>
```

**Comment by yc-codes** [comment]
Create a file in your project `.vscode/settings.json`

```json
{
    "tailwindCSS.classAttributes": [
        "class",
        "className",
        "ngClass",
        "enter",
        "enterFrom",
        "enterTo",
        "leave",
        "leaveFrom",
        "leaveTo",
    ]
}
```

**Reply by bibekjodd** [reply]
best solution
